akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1104,3 +1072,44 @@ private[stream] final class Reduce[T](f: (T, T) ⇒ T) extends SimpleLinearGraph
   }
   override def toString = "Reduce"
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] final class StatefulMapConcat[In, Out](f: () ⇒ In ⇒ immutable.Iterable[Out]) extends GraphStage[FlowShape[In, Out]] {
+  val in = Inlet[In]("StatefulMapConcat.in")
+  val out = Outlet[Out]("StatefulMapConcat.out")
+  override val shape = FlowShape(in, out)
+  override def initialAttributes: Attributes = DefaultAttributes.statefulMapConcat
+
+  def createLogic(inheritedAttributes: Attributes) = new GraphStageLogic(shape) {
+    var currentIterator: Iterator[Out] = Iterator.empty
```

Comment:
This reminds me: please put a `object EmptyIterator extends Iterator[Nothing]` into `akka.util.Collections` and use that here (as we as in EmptyImmutableSeq).